Tensile/KernelWriterAssembly.py

```diff
@@ -7314,27 +7327,24 @@ def localReadDo(self, kernel, bufferIdx, iui, epsi, tP):
 
             if kIdx % 2 == 1:
               # pack 2 half-dword into one
-              destVgpr = vgpr("Valu%s_X%u_I%u+%u" % (tc, bufferIdx, iui, int(valuIdx)))
-              imod.addInst("s_waitcnt lgkmcnt(0)", "")
               if kernel["MatrixInstruction"] and \
                  kernel["AssertSummationElementMultiple"] % kernel["MatrixInstK"] != 0 and \
                  self.inTailLoop:
-                imod.addInst("v_cmp_eq_u32", sgpr(isTailRemainderLoop, 2), sgpr("LoopCounter%s"%loopChar), 1, "check if last iter in tail loop")
-                imod.addInst("s_cmp_eq_u32", sgpr("NumRemainderSumElements%s"%loopChar), "0", "check if sum dim is divisible by k")
-                imod.addInst("s_cselect_b64", sgpr(isTailRemainderLoop, 2), "0x0", sgpr(isTailRemainderLoop, 2), "zero-fill only if last iter AND sum dim not divisible by k")
+                packCode.addInst("v_cmp_lt_u32", sgpr(isTailRemainderLoop, 2), sgpr("LoopCounter%s"%loopChar), kernel["MatrixInstK"], "check if last iter in tail loop")
```

Comment:
8 instructions for packing half-words..  do we have enough instruction slots for these packing instruction that we do not create bubbles in miSIMD pipeline

Reply:
In main loop body there's 1 extra instruction for packing every 2 half-dwords, plus the instruction interleaving @aazz44ss proposed in this patch so they are well hidden in the shadow of MIs

What you pointed out here are for the tail loop only (originally came from odd-k problem size support back in PR #809). Admittedly there's slight perf penalty. But again these extra instruction wouldn't be there if some size assertions are satisfied, i.e., `kernel["AssertSummationElementMultiple"] % kernel["MatrixInstK] == 0`